compiler/cpp/src/generate/t_php_generator.cc

```diff
@@ -1341,9 +1341,13 @@ void t_php_generator::generate_process_function(t_service* tservice, t_function*
     return;
   }
 
-  f_service_ << indent() << "$bin_accel = ($output instanceof "
-             << "TBinaryProtocolAccelerated) && function_exists('thrift_protocol_write_binary');"
+  // for compatibility, add method_exists
+  f_service_ << indent() << "$bin_accel = method_exists($output, 'isBinaryAccelerated')" << endl;
+  indent_up();
+  f_service_ << indent() << " && $output->isBinaryAccelerated()" << endl
+             << indent() << " && function_exists('thrift_protocol_write_binary');"
```

Comment:
The real problem may be that existing code generation here is too specific to one concrete type.
What if we move these thrift_protocol_..._binary calls to TBinaryProtocolAccelerated.php ?
That way we can move isStrict... checks there too so that we don't need them in TProtocolDecorator either.